### PR TITLE
Feature/create case

### PR DESCRIPTION
**Pull request by Charlie5991**

### CODE REVIEW

File: public/app/components/cases/cases.controller.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+(function() {
+    var mod = angular.module('cases', ['cases.create']);
+
+    mod.controller('CasesController', CasesController);
+
+    function CasesController($scope, $http, $modal) {
+
+        // $scope.itemsByPage = 10;
```
Comment (kkmah) [inline public/app/components/cases/cases.controller.js]: Remove unused/commented code

File: public/app/components/cases/cases.view.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                    </div>
+                </div>
+            </div>
+            <div class="panel-body">
+                <div class="table-responsive">
+                    <table datatable="ng" class="table table-striped table-bordered table-hover dataTables-example">
+                        <thead>
+                            <tr>
+                                <th>Name</th>
+                                <!-- <th>Email</th> -->
```
Comment (kkmah) [inline public/app/components/cases/cases.view.html]: Remove comments

File: public/app/components/cases/cases.view.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                                <!-- <th>Role</th> -->
+                                <!-- <th>Actions</th> -->
+                            </tr>
+                        </thead>
+                        <tbody>
+                            <tr ng-repeat="row in rowCollection">
+                                <td>{{row.name}}</td>
+                                <!-- <td>{{row.email}}</td>
+                                <td>{{row.role | capitalizeFirst}}</td>
```
Comment (kkmah) [inline public/app/components/cases/cases.view.html]: Remove comments

File: public/app/components/cases/create/create.controller.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            } else if (data.data.response.hits.total > 0) {
+                $scope.formData.users = data.data.response.hits.hits;
+                console.log(data.data.response.hits.hits);
+                users = data.data.response.hits.hits;
+            }
+        }, function errorCallback(data) {
+            toastr.error('Please try again later.', 'Something went wrong!');
+        });
+
+        // $http({
```
Comment (kkmah) [inline public/app/components/cases/create/create.controller.js]: More comments, please remove

File: public/app/components/cases/create/create.controller.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        //     // }
+        // });
+
+        $scope.save = function(cases){
+            var addCase = true;
+            console.log(cases);
+            // Save user
+            if(addCase) {
+                var tempCase = angular.copy(cases);
+                // delete tempCase.password2;
```
Comment (kkmah) [inline public/app/components/cases/create/create.controller.js]: You know the drill

File: public/app/components/cases/cases.view.html [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                    <a ui-sref="index">Home</a>
+                </li>
+                <li class="active">
+                    <strong>Cases</strong>
+                </li>
+            </ol>
+        </div>
+    </div>
+    <div class="wrapper wrapper-content animated fadeIn ng-scope">
+        <!-- Customers table -->
```
Comment (Moduo) [inline public/app/components/cases/cases.view.html]: Doesn't this suppose to be the `Cases table`?

File: public/app/components/cases/cases.controller.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
+(function() {
+    var mod = angular.module('cases', ['cases.create']);
+
+    mod.controller('CasesController', CasesController);
+
+    function CasesController($scope, $http, $modal) {
```
Comment (Moduo) [inline public/app/components/cases/cases.controller.js]: Do you really need to inject `$modal`? I don't see that you're using it.

File: public/app/components/cases/create/create.controller.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+                            if (typeof data.response !== 'undefined') {
+                                if (data.response.created == true) {
+                                    toastr.success('Case ' + tempCase.name +' has been added');
+                                    $state.go('cases', {}, { reload: true });
+                                }
+                            }
+                        }
+                    );
+            } else if (!addCase){
+                toastr.error('An user with this email address already exists');
```
Comment (Moduo) [inline public/app/components/cases/create/create.controller.js]: > An user
need to be 
> A user

File: routes/api.js [hunk 9]
```diff
@@ -983,6 +983,19 @@ router.get('/get_users', function(req, res){
...
 });
 
+router.get('/get_filtered_users', function(req, res){
+  var filter = req.body.filter;
+  c.elasticsearch.search({
+    index: INDEX,
+    type: 'users',
+    body: { query: { match: {'role': 'user'},}},
+    _source: ["id", "firstName", "lastName", "email", "isDeleted", "role"],
+    size: 999 // all
```
Comment (Moduo) [inline routes/api.js]: Does this mean that there can't be more than 999 users? I don't think 999 is all.

**Pull request merged**